### ISSUE TITLE
Bugs found in current build

### ISSUE THREAD

**Issue by XeenRecoil** [AppliedEnergistics/Applied-Energistics-2#684]
Hi;
First let me say that this mod is one of the best and I couldn't imagine Minecraft without it.

Now to the fun stuff, the bug list, I assume you are already aware of them but I wanted to let you know what I was experiencing anyway to confirm reproduction of the issue.

1) P2P causes huge server error logs to be created every day.
2) The Quantum Network Bridge sometimes does not load properly on server start up and requires a server restart to fix.
3) Unknown unstable network state occurs.
4) Security Terminal stops responding and enters an unrecoverable error state that requires reinstallation to fix.
5) Ender IO does not play nice with AE2.
These are currently the most problematic things I can think of.
Thanks for your consideration.

**Comment by Cisien** [comment]
1. Logs
2. Chunk load everything
3. Chunk load everything
4. Logs, setups, details.... SOMETHING!

**Comment by yueh** [comment]
Let me respond in a different order as it might be important

5) As you said, EnderIO is not playing nicely with AE2, but this is mainly because they do not support everything needed to work properly.
Using their conduits could probably result in any of your errors, so I would highly recommend to remove them from your networks and then test if any of your problems are still existing. If not it is a good idea to report your problems to EnderIO. Hopefully they remove themselves from the network and not linger around as part of the network until the mod is fully removed from your instance.

1) as @Cisien said, post the logs (not here but on pastebin or gist and just link them). Might be related to EnderIO.
2) QNBs are needing power until fully linked. This means when the server restarts and the chunks needed are loaded in a wrong order resulting them having no power and staying offline.
So either inject power on both sides or add some energy cells as buffer until everything is loaded.
3) Way to vague. More information, logs, etc. EnderIO?
4) EnderIO is known to break them, test again without it.

**Comment by XeenRecoil** [comment]
Thanks for the response guys, I know I didn't provide tons of information, but I do care about what happens with this mod because I think it's important to the game so I will give you as much information  as I have.
Ok so thinking that perhaps the issues might have been related to the QNB I consolidated all critical network components in the same chunk and the issue persists, channels are unstable and go offline and back online about 6 times a minute. 
I am still trouble shooting trying to figure this out and you will know what I know as soon as I know it.
Thanks

**Comment by XeenRecoil** [comment]
Update.
I moved the critical components to the same chunk and the issue persisted so I disconnected the QNB and the the issue ceased, I'm investigating to determine the root of the issue now.

**Comment by XeenRecoil** [comment]
Update.
Ok so I moved the Co-Processors/Storage units, Wireless access point and the Ore Quadrupling system to QNB #1 by themselves and all other terminals and machines to QNB #2 and the issue persisted so I disconnected QNB #2 and the issue was resolved.
Once I established that QNB #2 was the problem I investigated all devices connected to it and couldn't find anything wrong and after tripple checking everything I am stumped why it causes the issue.
I am at a stopping point because I don't know what else to check.

**Comment by XeenRecoil** [comment]
Ok so after checking all network connections on QNB #2 and eliminating machines from all the different mods one at a time I have come to the conclusion that the issue stems from having more than one QNB connected to the network at a time.
At least you guys know where to start looking now.

**Comment by XeenRecoil** [comment]
What I can't figure out is why the ME controller cares if there is a Bus in a chunk that isn't loaded and decides to reset because of it.

**Comment by XeenRecoil** [comment]
If there is ME cable in a chunk that isn't loaded in between the Controller or QNB and a Bus what happens?

**Comment by Cisien** [comment]
AE is a large multi-block. When part of I unloads, the 'network' reloads. This causes all devices attached to reset.

If something is causing a chunk to load and unload frequently, then you will see this behavior.

**Comment by XeenRecoil** [comment]
So after sleeping on the previous days events it occurs to me that this network reload behavior doesn't really need to happen at all.
The ME controller shouldn't care that there is a Terminal or Bus out there in no mans land somewhere that it can see but can't talk to, it should just say "I see you but I'm ignoring you, until I can talk to you" and go on it's merry way, this behavior would stop the annoying network reload issue.
The real issue is why would the network reset just because it finds a new device, shouldn't it just add it if there is an available channel without so much as a hickup?

**Comment by Cisien** [comment]
The current design of the ae grid forces it to rebuild the list of devices whenever it changes, this includes when chunks load and unload.

**Comment by yueh** [comment]
There is no alternative for it as every change to your network has the potential to take down the whole network, so it has to reevaluate the whole network. It might be possible to make the rebuild a bit more intelligent and more locally. But this would result in a pretty random behaviour as unloading cable A does not do anything at all and unloading B takes your network completely down. So a consistent behaviour is clearly the better approach.

The other option would be a static network, which would require to run back to the controller each time you change something and hit the button to rebuild it. Probably not what you want.

The issue with cables popping off is probably EnderIO related. As said the conduits are not working well currently and if you have a security station, they can force any adjacent ME cable to pop off.

**Comment by XeenRecoil** [comment]
It's definately not Ender IO related, because they pop off in locations where there is no ender IO cable within 20 meters.

**Comment by Mengmoshu** [comment]
@yueh How terrible would it be to have both the current setup and static networks as some sort of mode option, and a tool which you use to tell the network to rebuild when it's in static mode? Default to the current mode, and even provide a config to disable static networks?

As an aside, I can imagine a couple scenarios where clever use of chunk loaders and chunk boundaries could be useful. For example as a poor man's player detector to enable parts of the network that don't need to run all the time.

**Comment by yueh** [comment]
Pretty bad, it is one of the core functionalities. So more like maintaining two completely different AE version. Also it would only apply on adding parts, removing parts including unloaded chunks needs to be taken care of. Like putting whole network into an invalid state until reset.

It would change nothing besides being annoying and forcing a manual reset after each change.
Not chunkloaded parts would never be added to your network and forgetting it once after adding something might take your network down next reset. So no immediate feedback if something is wrong and you need to manually check your whole network to resolve it. But this is getting off topic.


@XeenRecoil are you using EnderIO ME conduits at all? I cannot remember currently how cables persist their owner, but the conduits might cause them to change the owner. Not loaded parts/chunks will still know their old one and then pop off on loading as they will by owned by someone else.

**Comment by Mengmoshu** [comment]
@yueh Thanks for the thorough answer. I hadn't thought the idea through completely.

**Comment by XeenRecoil** [comment]
I have continued to monitor the QNB not loading properly on server start up and it happens consistently and as far as I can tell isn't "user error" as a simple server restart solves the issue.

**Comment by Cisien** [comment]
On my server I have 8 qnbs spread all over the world, I have yet to see an issue where they fail to initialize on server start.

Any chance you could narrow down the case that causes this? (creative world with AE and required mods to repro)

**Comment by XeenRecoil** [comment]
I currently have the entire network powered with Ender IO Via a Capacitor bank through a green Energy Conduit to a Energy Acceptor to a Dense Energy Cell.
On the ME controller side of the QNB's (2) the Dense Energy Cell is directly interfaced to one of the ME controllers.
On the other side of the QNB's (2) the Dense Energy Cell is connected Via a Quartz Fiber to both QNB's indirectly Via a ME Dense cable.
I could redesign my entire power system and power the network with Thermal Expansion if you like but it would take some doing considering 90% of the ME hardware is at 25 elevation down on the bottom of a deep ocean floor.

**Comment by Cisien** [comment]
The mod providing power is likely not the problem, neither is AE power on either side of the qnb.

**Comment by XeenRecoil** [comment]
Are there server side operational logs generated by AE2, that could shed some light on this situation?

**Comment by thatcraniumguy** [comment]
Owner of the server where the issue was happening. Just came across this issue, and I'll be posting some logs later tonight.

Previously, we've been having issues with the P2P tunnels, where the error messages in the logs were spamming about 10x per second or more, and it was filling the logs. We were on rv1 earlier, but plan on trying out rv2 beta on a test server to see if further issues occur.

No errors have occured that I could see when the QNB fails to initialize, and from testing with the people on the server, it seems that even replacing the controllers on the network wouldn't help.

**Comment by Sperovita** [comment]
On the same server. The QNB did not reinitialize after a restart this time. I pulled the Security console and replaced it, still nothing. So I pulled the security console, the controllers, all the rings on both sides, then replaced them and it came back. Not just popping the center of the rings but the whole ring. Next time I'll try and narrow it down. Tried popping the center of the rings on both sides one after another before with no result.
